core/include/kernel/tee_common_otp.h

```diff
@@ -36,10 +36,31 @@ struct tee_hw_unique_key {
 	uint8_t data[HW_UNIQUE_KEY_LENGTH];
 };
 
-/* exposed to let tee_init set the key */
-extern uint8_t hw_key_digest[TEE_SHA256_HASH_SIZE];
+#if defined(CFG_OTP_SUPPORT)
 
 void tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey);
 int tee_otp_get_die_id(uint8_t *buffer, size_t len);
 
+#else
+
+#include <string.h>
```

Comment:
I have no strong opinion on this. Assuming we move this up, would you still keep the #ifdef guard, or include string.h unconditionally?
@pascal-brand-st-dev what do you prefer?